src/clm/commands/write_nn_Tc.py

```diff
@@ -24,29 +24,46 @@ def calculate_fingerprint(smile):
     return None
 
 
-def find_max_similarity_fingerprint(target_smile, ref_smiles, ref_fps):
-    target_fps = calculate_fingerprint(target_smile)
+def find_max_similarity_fingerprint(target_row, ref_smiles, ref_fps, ref_inchikeys):
```

Comment:
maybe add a `target_inchikey` argument after `target_row` and just have the caller pass the correct thing in.